v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/TextIOtoBigQueryLT.java

```diff
@@ -0,0 +1,178 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates;
+
+import static com.google.cloud.teleport.it.artifacts.ArtifactUtils.createStorageClient;
+import static com.google.cloud.teleport.it.artifacts.ArtifactUtils.getFullGcsPath;
+import static com.google.cloud.teleport.it.bigquery.BigQueryResourceManagerUtils.toTableSpec;
+import static com.google.cloud.teleport.it.matchers.TemplateAsserts.assertThatPipeline;
+import static com.google.cloud.teleport.it.matchers.TemplateAsserts.assertThatResult;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.teleport.it.DataGenerator;
+import com.google.cloud.teleport.it.TemplateLoadTestBase;
+import com.google.cloud.teleport.it.TestProperties;
+import com.google.cloud.teleport.it.artifacts.ArtifactClient;
+import com.google.cloud.teleport.it.artifacts.GcsArtifactClient;
+import com.google.cloud.teleport.it.bigquery.BigQueryResourceManager;
+import com.google.cloud.teleport.it.bigquery.DefaultBigQueryResourceManager;
+import com.google.cloud.teleport.it.common.ResourceManagerUtils;
+import com.google.cloud.teleport.it.launcher.PipelineLauncher.LaunchConfig;
+import com.google.cloud.teleport.it.launcher.PipelineLauncher.LaunchInfo;
+import com.google.cloud.teleport.it.launcher.PipelineOperator.Result;
+import com.google.cloud.teleport.metadata.TemplateLoadTest;
+import com.google.common.base.MoreObjects;
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.text.ParseException;
+import java.time.Duration;
+import java.util.function.Function;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Performance test for {@link TextIOToBigQuery TextIOToBigQuery} template. */
+@TemplateLoadTest(TextIOToBigQuery.class)
+@RunWith(JUnit4.class)
+public class TextIOtoBigQueryLT extends TemplateLoadTestBase {
+  private static final String SPEC_PATH =
+      MoreObjects.firstNonNull(
+          TestProperties.specPath(),
+          "gs://dataflow-templates/latest/flex/GCS_Text_to_BigQuery_Flex");
+  // 35,000,000 messages of the given schema make up approximately 10GB
+  private static final Long NUM_MESSAGES = 35000000L;
```

Comment:
Nit but `35_000_000L` is easier to read / avoid messing up with zeroes

Reply:
Done